api/envoy/api/v2/route/route.proto

```diff
@@ -920,6 +920,10 @@ message HeaderMatcher {
     // * For range [-10,0), route will match for header value -1, but not for 0, "somestring", 10.9,
     //   "-1somestring"
     envoy.type.Int64Range range_match = 6;
+
+    // If specified, header match will be performed based on if the header value is *not* the value
+    // specified or if the header is not present.
+    string inverse_match = 7;
```

Comment:
Instead of adding a new match type here, I would recommend adding a boolean value outside of the oneof called something like `invert_match` or something like that. Then you can just negate any of the match types. I think that would be more generally useful. WDYT?

Reply:
I think that's simpler to use and much more useful. 👍